Description: backport test_wclAutoCommitWordSetting from 5.12
Origin: upstream, https://code.qt.io/cgit/qt/qtvirtualkeyboard.git/commit/?id=6d429be4d7784121
Last-Update: 2018-09-29

--- a/tests/auto/inputpanel/data/inputpanel/inputpanel.qml
+++ b/tests/auto/inputpanel/data/inputpanel/inputpanel.qml
@@ -88,6 +88,7 @@ InputPanel {
     property alias soundEffectSpy: soundEffectSpy
     property alias inputMethodResultSpy: inputMethodResultSpy
     property alias wordCandidateListChangedSpy: wordCandidateListChangedSpy
+    property alias inputMethodSelectionListChangedSpy: inputMethodSelectionListChangedSpy
     property alias wordCandidateListVisibleSpy: wordCandidateListVisibleSpy
     property alias shiftStateSpy: shiftStateSpy
     property alias shadowInputControlVisibleSpy: shadowInputControlVisibleSpy
@@ -172,6 +173,12 @@ InputPanel {
     }
 
     SignalSpy {
+        id: inputMethodSelectionListChangedSpy
+        target: InputContext.inputEngine.inputMethod
+        signalName: "selectionListChanged"
+    }
+
+    SignalSpy {
         id: wordCandidateListVisibleSpy
         target: wordCandidateView
         signalName: "onVisibleConditionChanged"
--- a/tests/auto/inputpanel/data/tst_inputpanel.qml
+++ b/tests/auto/inputpanel/data/tst_inputpanel.qml
@@ -1677,19 +1677,14 @@ Rectangle {
                 skip("Prediction/spell correction not enabled")
 
             for (var len = 1; len <= 5; ++len) {
-                inputPanel.wordCandidateListChangedSpy.clear()
                 inputPanel.virtualKeyClick("z")
-                waitForRendering(inputPanel)
-                if (len >= 3) {
-                    if (data.wclAutoCommitWord)
-                        tryVerify(function() { return inputPanel.wordCandidateView.model.count === 0 }, 500)
-                    else
-                        wait(500)
+                if (len >= 2) {
+                    inputPanel.inputMethodSelectionListChangedSpy.clear()
+                    inputPanel.inputMethodSelectionListChangedSpy.wait()
                     if (inputPanel.wordCandidateView.model.count <= 1)
                         break
                 }
             }
-            waitForRendering(inputPanel)
 
             if (data.wclAutoCommitWord)
                 compare(inputPanel.wordCandidateView.model.count, 0)
